pkg/config/template.go

```diff
@@ -103,43 +108,53 @@ func (c *TemplateCMManager) CreateCMTemplates(templates CiTemplates) error {
 	return kutilerrors.NewAggregate(errors)
 }
 
-func (c *TemplateCMManager) CreateClusterProfiles(dir string, profiles []ClusterProfile) error {
-	var errs []error
-	for _, p := range profiles {
-		cm, err := genClusterProfileCM(dir, p)
-		if err != nil {
-			errs = append(errs, err)
-			continue
-		}
-		c.logger.WithFields(logrus.Fields{"cluster-profile": cm.ObjectMeta.Name}).Info("creating rehearsal cluster profile ConfigMap")
-		if err := c.createCM(cm); err != nil {
-			errs = append(errs, err)
-		}
-	}
-	return kutilerrors.NewAggregate(errs)
+type osFileGetter struct {
+	root string
 }
 
-func genClusterProfileCM(dir string, profile ClusterProfile) (*v1.ConfigMap, error) {
-	ret := &v1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{Name: profile.CMName()},
-		Data:       map[string]string{},
-	}
-	profilePath := filepath.Join(dir, profile.Name)
-	err := filepath.Walk(profilePath, func(path string, info os.FileInfo, err error) error {
-		if err != nil || info.IsDir() {
+func (g osFileGetter) GetFile(filename string) ([]byte, error) {
+	return ioutil.ReadFile(filepath.Join(g.root, filename))
+}
+
+func (c *TemplateCMManager) CreateClusterProfiles(dir string, cms map[string]prowplugins.ConfigMapSpec, profiles []ClusterProfile) error {
+	changes := []prowgithub.PullRequestChange{}
+	for _, profile := range profiles {
+		err := filepath.Walk(filepath.Join(dir, ClusterProfilesPath, profile.Name), func(path string, info os.FileInfo, err error) error {
+			if err != nil || info.IsDir() {
+				return err
+			}
+			if path, err = filepath.Rel(dir, path); err == nil {
+				changes = append(changes, prowgithub.PullRequestChange{
+					Filename: path,
+					Status:   string(prowgithub.PullRequestFileModified),
+				})
+			}
 			return err
```

Comment:
It's impossible for `filepath.Rel` to error here right? Might be worth a comment

Reply:
Added.